### DISCUSSION TITLE
Proposal: OpenBao Community Roles

### DISCUSSION THREAD

**Discussion by naphelps** [openbao/openbao#228] [Ideas]
This discussion is for a proposal to formalize community roles for OpenBao. This proposal will be included into OpenBao's future technical charter once the project becomes a formal Linux Foundation project.

How many roles does the community need (1-3, less/more)?
What permissions does each role require?
What are the requirements for gaining/losing the role?

- Role: ?

- Role: Contributor
  - Requirements:
    - ?? 
  - Permissions:
    - Issue Triage. 

- Role: Maintainer
  - Requirements:
    - Has made a pull request contribution.
    - Active in community.
    - Requires TSC vote.
  - Permissions:
    - Issue Triage
    - Pull Request Reviews
    - Pull Request Merge
    - Discussion moderation
    - Manage GitHub Actions/Workflows
    - Backups for community calls

**Comment by cipherboy** [comment]
I finally got around to looking at this, and while I don't think we're quite there yet, I think something like [Notary's split-level ownership](https://github.com/notaryproject/.github/blob/main/GOVERNANCE.md), which I think is fitting as it is similarly structured (core + plugins). I'd perhaps suggest a "plugin maintainer" role (given rights only on a limited set of repos) versus an "org maintainer" role (given rights on the main repo and all plugins).

Otherwise, I think this looks fine :-)  I like the idea of delegating community contributor status to give issue triage and perhaps also moderation capabilities (hide off-topic comments &c). 

Thoughts?

**Comment by cipherboy** [comment]
With the merge of #568  and #572, I'm happy to announce the proposal has been accepted and this can be closed!
